nifi-nar-bundles/nifi-snmp-bundle/nifi-snmp-processors/src/main/java/org/apache/nifi/snmp/utils/UsmReader.java

```diff
@@ -0,0 +1,145 @@
+import org.apache.nifi.processor.exception.ProcessException;
+import org.snmp4j.security.UsmUser;
+import org.snmp4j.smi.OID;
+import org.snmp4j.smi.OctetString;
+
+import java.io.File;
+import java.io.FileNotFoundException;
+import java.io.IOException;
+import java.util.Arrays;
+import java.util.List;
+import java.util.Scanner;
+import java.util.stream.Collectors;
+
+public interface UsmReader {
+
+    List<UsmUser> readUsm(String usmUsers);
+
+    static UsmReader jsonFileUsmReader() {
+        return usmUsersFilePath -> {
+            final List<UsmUser> userDetails;
+            try (Scanner scanner = new Scanner(new File(usmUsersFilePath))) {
+                final String content = scanner.useDelimiter("\\Z").next();
+                ObjectMapper mapper = new ObjectMapper();
+                SimpleModule module = new SimpleModule();
+                module.addDeserializer(UsmUser.class, new UsmUserDeserializer());
+                mapper.registerModule(module);
+                userDetails = mapper.readValue(content, new TypeReference<List<UsmUser>>() {
+                });
+            } catch (FileNotFoundException e) {
+                throw new ProcessException("USM user file not found, please check the file path and file permissions.", e);
+            } catch (JsonProcessingException e) {
+                throw new ProcessException("Could not parse USM user file, please check the processor details for examples.", e);
+            }
+            return userDetails;
+        };
+    }
+
+    static UsmReader jsonUsmReader() {
+        return usmUsersJson -> {
+            try {
+                ObjectMapper mapper = new ObjectMapper();
+                SimpleModule module = new SimpleModule();
+                module.addDeserializer(UsmUser.class, new UsmUserDeserializer());
+                mapper.registerModule(module);
+                return mapper.readValue(usmUsersJson, new TypeReference<List<UsmUser>>() {
+                });
+            } catch (JsonProcessingException e) {
+                throw new ProcessException("Could not parse USM user file, please check the processor details for examples.", e);
+            }
+        };
+    }
+
+    static UsmReader securityNamesUsmReader() {
+        return usmSecurityNames -> Arrays.stream(usmSecurityNames.trim().split(","))
+                .map(securityName -> new UsmUser(
+                        new OctetString(securityName), null, null, null, null)
+                )
+                .collect(Collectors.toList());
+    }
+
+    class UsmUserDeserializer extends StdDeserializer<UsmUser> {
+
+        public UsmUserDeserializer() {
+            this(null);
+        }
+
+        public UsmUserDeserializer(Class<?> vc) {
+            super(vc);
+        }
+
+        @Override
+        public UsmUser deserialize(JsonParser jp, DeserializationContext ctxt) throws IOException {
+            JsonNode node = jp.getCodec().readTree(jp);

```

Comment:
As we agreed, this can be moved to validation, since we know this information during the processor configuration and expression language is not supported for these fields.